### PR TITLE
[Settings] Moved changedisplayname to usercontroller

### PR DESCRIPTION
**Pull request by rullzer**
Killed the old static route to change a users display name and moved it to a properly testable controller.
- Unit test added

CC: @PVince81 @MorrisJobke @nickvergessen @LukasReschke

### CODE REVIEW

File: settings/controller/userscontroller.php [hunk 1]
```diff
@@ -585,4 +585,57 @@ public function stats() {
...
+			$username = $currentUser->getUID();
+		}
+
+		$user = $this->userManager->get($username);
+
+		if ($user === null ||
+			(
+				!$this->groupManager->isAdmin($currentUser->getUID()) &&
+				!$this->groupManager->getSubAdmin()->isUserAccessible($currentUser, $user) &&
+				$currentUser !== $user)
```
Comment (LukasReschke) [inline settings/controller/userscontroller.php]: Can we also add a check for `\OCP\IUser::canChangeDisplayName`?

Comment (rullzer) [inline settings/controller/userscontroller.php]: sure we can... but I assume that is already checked when we try to change it as well.

Comment (rullzer) [inline settings/controller/userscontroller.php]: Yes it is checked in https://github.com/owncloud/core/blob/master/lib/private/user/user.php#L140 and thus handled in https://github.com/owncloud/core/pull/20559/files#diff-86a88ac566bd79dc95498ce546f67a78R631

Comment (LukasReschke) [inline settings/controller/userscontroller.php]: :+1:

**Comment by PVince81** [comment]
Tested, has an issue regarding the restriction.
- [x] TEST: change display name in personal page still works
- [x] TEST: change display name in users page still works
- [ ] TEST: change display name in personal page when not allowed :no_entry_sign:  FAIL

1. Allow users to change display name
1. Go to personal page
1. Change config.php to disallow users to change display name
1. Now still on the personal page where the field is still there, enter a display name, wait for save
1. Refresh the page

Expected: display name not changed (controller must return error)
Actual: display name DID change

**Comment by rullzer** [comment]
@PVince81 fixed

**Comment by PVince81** [comment]
:+1:

**Comment by rullzer** [comment]
see https://github.com/owncloud/core/pull/20559#discussion_r45071534 for the :+1: from @LukasReschke

**Pull request merged**